src/components/Compare/CompareComponent.tsx

```diff
@@ -0,0 +1,380 @@
+        });
+    }
+
+    @autobind
+    private toggleNotEqualEntires() {
+        this.setState({
+            diffFilter : !this.state.diffFilter
+        });
+    }
+
+    @autobind
+    private toggleMissingEntires() {
+        this.setState({
+            missingFilter : !this.state.missingFilter
+        });
+    }
+
+    private gridComponentWidth(width) {
+        return width / 2 - 25;
+    }
+
+    private gridComponentHeight(height) {
+        return height - 50;
+    }
+
+    public render() {
+        return (
+            <div className={'compare-base-container'}>
+            <AutoSizer onResize={this.onGridResize}>
+                {({ height, width }) => (
+                        <ScrollSync>
+                            {({ onScroll, scrollLeft, scrollTop }) => (
+                                <div className={'compare-resize-component'} style={{ width, height }}>
+                                    <Button
+                                        className={classNames({'compare-button-active' : this.state.missingFilter})}
+                                        onClick={this.toggleMissingEntires}
+                                        title={'Show/Hide equal entires'}
+                                        icon={' icon-arrows'}
+                                    />
+                                    <Button
+                                        className={classNames({'compare-button-active' : this.state.diffFilter})}
+                                        onClick={this.toggleNotEqualEntires}
+                                        title={'Show/Hide equal entires'}
+                                        icon={'icon-not_equal'}
+                                    />
+                                    <Button
+                                        className={classNames({'compare-button-active' : this.state.equalFilter})}
+                                        onClick={this.toggleEqualEntires}
+                                        title={'Show/Hide equal entires'}
+                                        icon={'icon-equal'}
+                                    />
+                                    <div className={'grid-component-container'} style={{height : height - 50}}>
+                                        <div className={'compare-single-grid-wrapper'}>
+                                            <GridHeader
+                                                allColumns={this.columns}
+                                                headerColumns={this.columns}
+                                                columnWidths={this.state.columnWidths}
+                                                width={this.gridComponentWidth(width)}
+                                                scrollLeft={scrollLeft}
+                                                className={'grid-component-header'}
+                                                groupBy={[]}
```

Comment:
this cause rerender always. Maybe ommit this prop or use const?

Reply:
Made groupBy optional in QuickGridHeader as suggested.